src/main/kotlin/com/hibob/academy/dao/PetDao.kt

```diff
@@ -0,0 +1,74 @@
+package com.hibob.academy.dao
+
+import org.jooq.DSLContext
+import org.jooq.Record
+import org.jooq.RecordMapper
+import org.jooq.impl.DSL
+import org.springframework.stereotype.Component
+import java.math.BigInteger
+import java.time.LocalDateTime
+import java.sql.Date
+
+
+data class Pet(val name: String, val type: String, val dateOfArivval: Date, val companyId: Long , val ownerId: Long )
+
+data class PetWithOutType(val name: String, val dateOfArivval: Date, val companyId: Long)
+
+@Component
+class PetDao(private val sql: DSLContext ) {
+    private val table = PetTable.instance
+
+    private val petWithOutTypeMapper = RecordMapper<Record, PetWithOutType> { record ->
+        PetWithOutType(record[table.name], record[table.dateOfArrival], record[table.companyId])
+    }
```

Comment:
Why do we need this mapper?

Reply:
because in the exercise in jooq read - Create a function that retrieve all the pets by a given type  and return their name, dateOfArrival and company Id as a new data class. there are some values that are exist in pet but not in the values that return so i did another data class that specific to this .  what is the right way to do this ? or is it ok to return a pet with all the values - also the ownerId and the id ?